### PR TITLE
whitesur-wallpapers: init at 1.0

### PR DESCRIPTION
**Pull request by sei40kr**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Kranzes — COMMENTED** [review]
I don't think wallpapers belong as packages inside nixpkgs. Wallpapers increase the closure size of the channels. If you want to manage your wallpapers with nix you should use fetchurl directly from your config or add it as a package to your own personal config since wallpaper don't benefit from the binary cache provided for nixpkgs.

This package will be over 115MB of just `.png` and `.jpg` files...

File: pkgs/data/misc/whitesur-wallpapers/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  src = fetchFromGitHub {
+    owner = "vinceliuice";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-76oxSsjK5vuGgb1wQOEXNi/76e9yqHQoXxANqzrkbl8=";
+  };
+
+  nativeBuildInputs = [ bash ];
+
```
Comment (SuperSandro2000) [inline pkgs/data/misc/whitesur-wallpapers/default.nix:15]:
```suggestion
```
part of stdenv

File: pkgs/data/misc/whitesur-wallpapers/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ bash, fetchFromGitHub, lib, stdenvNoCC }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "WhiteSur-wallpapers";
```
Comment (romildo) [inline pkgs/data/misc/whitesur-wallpapers/default.nix:4]:
```suggestion
  pname = "whitesur-wallpapers";
```

**Comment by iblech** [comment]
> I don't think wallpapers belong as packages inside nixpkgs. Wallpapers increase the closure size of the channels. If you want to manage your wallpapers with nix you should use fetchurl directly from your config or add it as a package to your own personal config since wallpaper don't benefit from the binary cache provided for nixpkgs.
> 
> This package will be over 115MB of just `.png` and `.jpg` files...

On the one hand, I tend to agree: The closure size of the full channel is already huge.

On the other hand, we already have other wallpaper collections (adapta-backgrounds, fedora-backgrounds), the increase in closure size is small compared to the total closure size of the full channel, and most importantly: We have an issue with discoverability, right? If @sei40kr keeps this code to themselves, or pushes it somewhere on GitHub, others are likely to reproduce their efforts.

**Comment by RaitoBezarius** [comment]
I am not in favor of adding wallpapers in the repository, this is a good use case IMHO for a NUR oriented for wallpapers or something.

**Comment by SuperSandro2000** [comment]
Closing for the above reason and because it is stale.

**Pull request closed without merging**